mtd: m25p80: Add support for Macronix 25L8005

Add support for Macronix 25L8005.  Tested on a HP t5325 Thin Client.

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>
[bwh: Adjust for 2.6.35]

--- a/drivers/mtd/devices/m25p80.c	2010-06-19 10:37:08.000000000 +0000
+++ b/drivers/mtd/devices/m25p80.c	2010-06-19 10:37:54.000000000 +0000
@@ -621,6 +621,7 @@
 
 	/* Macronix */
 	{ "mx25l4005a",  INFO(0xc22013, 0, 64 * 1024,   8, SECT_4K) },
+	{ "mx25l8005",   INFO(0xc22014, 0, 64 * 1024,  16, SECT_4K) },
 	{ "mx25l3205d",  INFO(0xc22016, 0, 64 * 1024,  64, 0) },
 	{ "mx25l6405d",  INFO(0xc22017, 0, 64 * 1024, 128, 0) },
 	{ "mx25l12805d", INFO(0xc22018, 0, 64 * 1024, 256, 0) },
